src/pages/nft/class/_classId/index.vue

```diff
@@ -888,14 +883,42 @@ export default {
         };
         logPurchaseFlowEvent(this, 'add_to_cart', purchaseEventParams);
         logPurchaseFlowEvent(this, 'begin_checkout', purchaseEventParams);
-        const gaClientId = await getGaClientId(this);
-        const link = getNFTBookPurchaseLink({
-          classId: this.classId,
-          priceIndex: edition.index,
-          platform: this.platform,
-          gaClientId,
-        });
-        window.open(link, '_blank', 'noopener');
+        if (edition.price === 0) {
+          this.$router.push(
+            this.localeLocation({
+              name: 'nft-claim',
+              query: {
+                class_id: this.classId,
+                type: 'nft_book',
+                free: true,
+                price_index: edition.index,
+                from: 'liker_land_waived',
+              },
+            })
+          );
+          return;
+        }
+        if (edition.price > 0) {
+          await this.fetchNFTPrices();
```

Comment:
Should fetch when edition is changed instead of onClick, adding too much delay in user action might cause issue in popup/redirect